torch/csrc/jit/codegen/cuda/fusion_segmenter.cpp

```diff
@@ -3051,33 +3052,41 @@ void SegmentedFusion::setCachedHeuristicDataFor(
 namespace {
 
 //! A thin traversal class that collects all the tensorviews
-//!  that could cast to fp16 if they were segmented edges.
+//!  that could cast to fp16 or bf16 if they were segmented edges.
 //!  The selected values are currently defined as all the
 //!  tensorviews that
 //!     1. are not complete fusion input/output,
 //!     2. have a use chain that ends with a fp16
 //!         complete fusion output
 //!     3. are fp32 datatype
-class ForceFP16Annotation : public IterVisitor {
+class ForceHalfAnnotation : public IterVisitor {
  public:
-  static std::unordered_set<TensorView*> getAnnotatedSet(Fusion* fusion) {
-    ForceFP16Annotation annotation;
+  static std::unordered_set<TensorView*> getFP16AnnotatedSet(Fusion* fusion) {
+    ForceHalfAnnotation annotation;
     std::vector<Val*> fp16_outputs;
 
     std::copy_if(
         fusion->outputs().begin(),
         fusion->outputs().end(),
         std::back_inserter(fp16_outputs),
         [](auto* val) {
```

Comment:
I think we need to capture `cast_to_type_` here, i.e. ` [&cast_to_type_ ](auto* val) {`

Did it compile without it?